### ISSUE TITLE
Upsert (Update or Insert) Person Record

### ISSUE THREAD

**Issue by nelsonic** [dwyl/auth#53]
At present we are inserting an initial person into `people` in `seeds.exs` based on the `ADMIN_EMAIL` environment variable. This person is the (_first_) "admin" of the auth app.
They will have full control over the auth app. The issue is that we don't want to hard-code too much data into the `seed.exs` and we don't want to have 5 environment variables for defining the Admin person. 
So when the Admin logs into the App for the first time, they see this:
![image](https://user-images.githubusercontent.com/194400/80393260-27182b00-88a8-11ea-9565-4e2726c6204b.png)

When it _should_ show this:
![image](https://user-images.githubusercontent.com/194400/80393350-4616bd00-88a8-11ea-8386-13a60372b836.png)

We need to update the person record for the Admin when they successfully login with their Google Account.

With that in mind we need an `upsert_person/1` function.
This will be useful for anyone who updates their Google/GitHub profile e.g: their profile picture and wants that change to flow through to their @dwyl profile. 

# Todo

+ [x] Create an `upsert_person/1` function that accepts a `person` record and either creates a new record if one does not already exist or updates the existing record with the new values.

**Comment by nelsonic** [comment]
This is what I've got:
```elixir
  def upsert_person(person) do
    case get_person_by_email(person.email) do
      nil ->
        create_person(person)

      ep -> # existing person
        merged = Map.merge(AuthPlug.Helpers.strip_struct_metadata(ep), person)
        {:ok, person} = changeset(%Person{id: ep.id}, merged)
        |> Repo.update()

        person
    end
  end
```
Does it seem reasonable to you?

**Comment by nelsonic** [comment]
The `upeser_person/1` function works to update the "Admin" profile on first login:
![image](https://user-images.githubusercontent.com/194400/81042922-233c6800-8ea9-11ea-9da9-5693f11ca358.png)

This is included/used in #43
